src/filters/filter-fields-component.js

```diff
@@ -40,42 +41,49 @@ angular.module('patternfly.filters').component('pfFilterFields', {
       });
     };
 
-    ctrl.$postLink = function () {
-      $scope.$watch('config', function () {
+    ctrl.$onChanges = function () {
+      setupConfig ();
+    };
+
+    ctrl.$doCheck = function () {
+      // do a deep compare on config
+      if (!angular.equals(ctrl.config, prevConfig)) {
         setupConfig();
-      }, true);
+      }
     };
 
     function selectField (item) {
       ctrl.currentField = item;
-      ctrl.config.currentValue = null;
+      ctrl.currentValue = null;
     }
 
     function selectValue (filterValue) {
       if (angular.isDefined(filterValue)) {
-        ctrl.addFilterFn(scope.currentField, filterValue);
-        ctrl.config.currentValue = null;
+        ctrl.addFilterFn(ctrl.currentField, filterValue);
+        ctrl.currentValue = null;
       }
     }
 
     function onValueKeyPress (keyEvent) {
       if (keyEvent.which === 13) {
-        ctrl.addFilterFn(ctrl.currentField, ctrl.config.currentValue);
-        ctrl.config.currentValue = undefined;
+        ctrl.addFilterFn(ctrl.currentField, ctrl.currentValue);
+        ctrl.currentValue = undefined;
       }
     }
 
     function setupConfig () {
-      if (ctrl.fields === undefined) {
-        ctrl.fields = [];
+      prevConfig = ctrl.config;
```

Comment:
I think you need to copy the current ctrl.config, else prevConfig will always equal ctrl.config as ctrl.config gets updated.  Ie.   `prevConfig = angular.copy(ctrl.config);`